pyproject.toml

```diff
@@ -61,6 +61,14 @@ PyPDF2 = "*"
 pylatexenc = "*"
 python-pptx = "*"
 opencv-python = "*"
+litellm = "*"
+diskcache = "5.6.3"
+grep-ast = "0.3.2"
+tree-sitter = "0.21.3"
+tree-sitter-languages = "1.10.2"
+gitpython = "*"
+networkx = "*"
+scipy = "*"
```

Comment:
Are all of these dependencies necessary?

Reply:
Yes - it's required to pass the unit tests